content/en/blog/_posts/2021-12-15-pod-security-admission-beta.md

```diff
@@ -0,0 +1,612 @@
+Annotations:  <none>
+Status:       Active
+
+No resource quota.
+
+No LimitRange resource.
+```
+
+Let's create a new namespace and see if the labels apply there.
+
+```shell
+kubectl create namespace test-defaults
+kubectl describe namespace test-defaults
+```
+
+Same.
+
+```
+Name:         test-defaults
+Labels:       kubernetes.io/metadata.name=test-defaults
+Annotations:  <none>
+Status:       Active
+
+No resource quota.
+
+No LimitRange resource.
+```
+
+Can a privileged workload be deployed?
+
+```shell
+cat <<EOF | kubectl -n test-defaults apply -f -
+apiVersion: v1
+kind: Pod
+metadata:
+  name: pause-privileged
+spec:
+  containers:
+  - name: pause
+    image: k8s.gcr.io/pause
+    securityContext:
+      allowPrivilegeEscalation: true
+EOF
+```
+
+Hmm... yep. The default `warn` level is working at least.
+
+```
+Warning: would violate PodSecurity "restricted:latest": allowPrivilegeEscalation != false (container "pause" must set securityContext.allowPrivilegeEscalation=false), unrestricted capabilities (container "pause" must set securityContext.capabilities.drop=["ALL"]), runAsNonRoot != true (pod or container "pause" must set securityContext.runAsNonRoot=true), seccompProfile (pod or container "pause" must set securityContext.seccompProfile.type to "RuntimeDefault" or "Localhost")
+pod/pause-privileged created
+```
+
+Let's delete the pod with `kubectl -n test-defaults delete pod/pause-privileged`.
+
+Is my config even working?
+
+```shell
+# if using docker, shell into the control plane
+docker exec -it kind-control-plane bash
+
+# cat out the file we mounted
+cat /etc/kubernetes/policies/pod-security.yaml
+
+# check the api server logs
+cat /var/log/containers/kube-apiserver*.log 
+
+# check the api server config
+cat /etc/kubernetes/manifests/kube-apiserver.yaml 
+```
+
+**UPDATE:** The baseline policy permits `allowPrivilegeEscalation`. While I cannot see the Pod Security default levels of enforcement, they are there. Let's try to provide a manifest that violates the baseline by requesting hostNetwork access.
+
+```shell
+cat <<EOF | kubectl -n test-defaults apply -f -
+apiVersion: v1
+kind: Pod
+metadata:
+  name: pause-privileged
+spec:
+  containers:
+  - name: pause
+    image: k8s.gcr.io/pause
+  hostNetwork: true
+EOF
+```
+
+The output is similar to this:
+
+```shell
+Error from server (Forbidden): error when creating "STDIN": pods "pause-privileged" is forbidden: violates PodSecurity "baseline:latest": host namespaces (hostNetwork=true)
+```
+
+#### Yes!!! It worked! 🎉🎉🎉
+
+I later found out, another way to check if things are operating as intended is to check the raw API server metrics endpoint.
+
+Run the following command:
+
+```
+kubectl get --raw /metrics | grep pod_security_evaluations_total
+```
+
+The output is similar to this:
+
+```
+# HELP pod_security_evaluations_total [ALPHA] Number of policy evaluations that occurred, not counting ignored or exempt requests.
```

Comment:
Is this alpha specifically for the `pod_security_evaluations_total ` metric?

Reply:
cc @liggitt who shared the `metrics` pro-tip with me.

Reply:
yes, the metric is alpha (metrics currently only have ALPHA or STABLE stability levels)